assignments/arrays.js

```diff
@@ -80,17 +84,36 @@ console.log();
 // ==== Challenge 4 ====
 // The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
 let carYears = [];
-console.log();
+
+    for(i = 0; i < inventory.length; i++) {
+        carYears.push(inventory[i].car_year);
+    }
+console.log(carYears);
 
 // ==== Challenge 5 ====
 // The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
 let oldCars =[];
-console.log(); 
+
+    for(i = 0; i < inventory.length; i++) {
+        if(inventory[i].car_year > 2000) {
+            oldCars.push(inventory[i].car_year);
+        }
+
+    }
+
+console.log(oldCars.length); 
 
 // ==== Challenge 6 ====
 // A buyer is interested in seeing only BMW and Audi cars within the inventory.  Return an array that only contains BMW and Audi cars.  Once you have populated the BMWAndAudi array, use JSON.stringify() to show the results of the array in the console.
 let BMWAndAudi =[];
-console.log();
+
+    for(i = 0; i < inventory.length; i++) {
+        if((inventory[i].car_make === "BMW") || (inventory[i].car_make === "AUDI")) {
+            BMWAndAudi.push(inventory[i].car_make);
+        }
+    }
+
+console.log(BMWAndAudi);
 
```

Comment:
The instructions ask you to log the entire object, provided that the make is either a 'BMW' or an 'Audi'. Here, you are only logging the make of the cars.  Also, I believe that your conditional statement is case-sensitive, so having AUDI in all caps skews your results. Then, it is asking you to use the JSON.stringify method to parse your objects and turn them into strings. Here is how that would look:

```suggestion
for(i = 0; i < inventory.length; i++) {     
        if((inventory[i].car_make === "BMW") || (inventory[i].car_make === "Audi")) {      
            BMWAndAudi.push(inventory[i]);       
        }
    }

 console.log(JSON.stringify(BMWAndAudi));
```